source/includes/library-reference/_library-reference.html.md

```diff
@@ -0,0 +1,180 @@
+message sequence sent from the AI Engine between training and test passes. See
+the [configure clause of the lesson statement][1] in this simulator's
+accompanying curriculums to get more information about the values in `kwargs`.
+
+
+## advance(actions)
+
+```python
+    def advance(self, actions):
+        """ Function to make a move based on output from the BRAIN as
+        defined in the Inkling file.
+        """
+        if self.value > self.max or self.value < self.min:
+            reset()
+        self.value += actions["delta"]
+```
+
+Advance the simulation forward one tick. `actions` contains a dictionary of key
+values as defined by this simulator's action schema in Inkling.
+
+
+## get_state()
+
+```python
+    def get_state(self):
+        """ Gets the state of the simulator, whether it be a valid value or
+            terminal ("game over") state.
+        """
+        return SimState(
+                 state={"value": self.value, "min": self.min, "max": self.max},
+                 is_terminal=self.value > self.max or self.value < self.min)
+```
+
+Returns a [SimState][2] containing the state of the simulation computed after
+the last call to `advance`. `SimState` contains a `state` and an `is_terminal`
+flag. `state` is a dictionary matching the state schema as defined in an
+Inkling `simulator` statement within an Inkling program. `is_terminal` is only
+true when the simulator is in a "game over" state. Any cleanup of the
+`is_terminal` state should be handled by `advance` or `get_state`.
+
+
+## Objectives (Reward Functions)
+
+```python
+    def distance_to_goal(self):
+        """ Distance between the current value, as adjusted by the BRAIN and
+            the desired value (the mean between min and max)
+        """
+        goal = int((self.max + self.min)/2)
+        return abs(goal - self.value)
+```
+
+Any methods added to this class that are not named with one of the above
+method names are eligible to be called as a reward function.  To be called,
+the name of the function specified in an Inkling `lesson` must match the name
+of the method.  Reward functions specified in Inkling but not implemented on a
+`Simulator` class will result in an error.
+
+## SimState
+
+A named tuple returned by `Simulator.get_state`.
+
+
+### state
+
+A dictionary matching the state schema as defined in an Inkling `simulator`
+statement within an Inkling program.
+
+
+### is_terminal
+
+`True` when the simulator is in a "game over" state; otherwise `False`.
+
+
+# Running the Simulator
+
+## run_for_training_or_prediction(name, simulator_or_generator)
+
+Connects to the AI Engine and identifies as `name`. The `name` of this
+simulator must be the same in Inkling as it is in this function call.
+`simulator_or_generator` is a reference to an instance of the simulator. This
+call blocks and drives the simulator loop until training stops.
+
+
+# Example
+
+For more information, see [examples][3].

```

Comment:
Maybe a little more verbiage here to say something like "the complete example of Find the Center, from which the previous code snippets were pulled from, can be found on our Examples[3] page."